lightkurve/seismology/numax_estimators.py

```diff
@@ -0,0 +1,155 @@
+"""Helper functions for estimating numax from periodograms.
+
+Functions in this module should be named "estimate_numax_methodname()".
+"""
+import numpy as np
+from matplotlib import pyplot as plt
+
+from astropy.convolution import convolve, Box1DKernel, Gaussian1DKernel
+from astropy import units as u
+
+from .. import MPLSTYLE
+from . import utils
+from .utils import SeismologyQuantity
+
+
+def estimate_numax_acf(periodogram, numaxs=None, window=None, spacing=None):
+    """Estimate the frequency at maximum power from a periodogram using
+    the autocorrelation method.
+    """
+    # Calculate the window size
+
+    #C: What is this doing? Why have these values been picked? This function is slow.
+    if window is None:
+        if u.Quantity(periodogram.frequency[-1], u.microhertz) > u.Quantity(500., u.microhertz):
+            window = u.Quantity(250., u.microhertz).to(periodogram.frequency.unit).value
+        else:
+            window = u.Quantity(25., u.microhertz).to(periodogram.frequency.unit).value
+
+    # Calculate the spacing size
+    if spacing is None:
+        if u.Quantity(periodogram.frequency[-1], u.microhertz) > u.Quantity(500., u.microhertz):
+            spacing = u.Quantity(10., u.microhertz).to(periodogram.frequency.unit).value
+        else:
+            spacing = u.Quantity(1., u.microhertz).to(periodogram.frequency.unit).value
+
+    # Run some checks on the inputs
+    window = u.Quantity(window, periodogram.frequency.unit).value
+    spacing = u.Quantity(spacing, periodogram.frequency.unit).value
+    if numaxs is None:
+        numaxs = np.arange(np.ceil(np.nanmin(periodogram.frequency.value)) + window/2,
+                    np.floor(np.nanmax(periodogram.frequency.value)) - window/2,
+                    spacing)
+    numaxs = u.Quantity(numaxs, periodogram.frequency.unit).value
+    if not hasattr(numaxs, '__iter__'):
+        numaxs = np.asarray([numaxs])
+
+    fs = np.median(np.diff(periodogram.frequency.value))
+    for var, label in zip([np.asarray(window), np.asarray(spacing), numaxs],
+                          ['window', 'spacing', 'numaxs']):
+        if (var < fs).any():
+            raise ValueError("You can't have {} smaller than the "
+                            "frequency separation!".format(label))
+        if (var > (periodogram.frequency[-1].value - periodogram.frequency[0].value)).any():
```

Comment:
This check doesn't work for `numaxs` when running this code on a periodogram that has been sliced (ie not running from 0 to Nyquist). I'm not sure why this is passing the unit tests on sliced periodograms--- it shouldn't be.

Btw this series of checks is incredibly elegant, I love it!!